core/controllers/admin.py

```diff
@@ -450,11 +460,13 @@ def post(self) -> None:
                         'The \'commit_message\' must be provided when the '
                         'action is update_platform_parameter_rules.'
                     )
+                default_value = self.normalized_payload.get('default_value')
 
                 try:
                     registry.Registry.update_platform_parameter(
                         platform_param_name, self.user_id, commit_message,
-                        new_rules
+                        new_rules,
+                        default_value['value']
```

Comment:
The following might be the reasons for the error you are facing:

1. In the schema, you are defining `default_value`'s default value as `None` but you are not ensuring to the code that this value is not going to be None from line 463 (where you are fetching the `default_value` from `normalized_payload`) till here, which causes MyPy to throw Optional[] related error.
2. In your schema, you have defined `default_value`'s type as object_dict whereas in your `TypedDict` you have defined it as `parameter_domain.PlatformDataTypes`. So, that's a contradiction that MyPy is facing. Because object_dicts are always considered as dictionaries but you have typed it as Union type. To fix this, I would suggest typing your default_value in `TypedDict` as Dict type.

Reply:
Thanks @sahiljoster32 for the help. The error is resolved now.